llvm-bazel/zlib_disable.BUILD

```diff
@@ -0,0 +1,10 @@
+# This file is licensed under the Apache License v2.0 with LLVM Exceptions.
```

Comment:
I think these should go under third_party_build to tuck them away a bit more. Alternatively, there's enough files here that we should maybe have a zlib directory.

Reply:
I thought about `thrid_party_build` but that seems wrong: this doesn't actually build any third-party code, it is a wrapper only.

I'm happy to create a directory if you want, but didn't seem quite worth it (4 files) to me.

Reply:
I named the directory that for "third-party build files" :grin: The first thing it contained was a build file for zlib. We can totally call that directory something else. `dependency_configuration`? But with zlib and terminfo up here it seems like a lot of files

Reply:
Some discussion in discord. We're going to put "real" build files that actually compile some third-party source code (and therefore should carry a license header) in `third_party_build` and put implementation details of dependency configuration (like build files with empty rules) in `deps_impl`. We'll leave the bzl configurations in `llvm-bazel` as they are actually "interesting"

Reply:
As discussed on chat, we have `third_party_build` for rules that actually build third-party code (and thus maybe need a license).

These bits are just implementation details and don't need that. Moved them to a `deps_impl` directory to make it clear.